File: src/main/java/org/grandviewtech/entity/enums/NoNc.java
package org.grandviewtech.entity.enums;

public enum NoNc
	{
	NO("NO"), NC("NC") ,DEFAULT("DEFAULT") ,SET("SET"), RESET("RESET");
		
		private String type;
		
		private NoNc(String type)
			{
				this.type = type;
			}
			
		public String getType()
			{
				return type;
			}
	}

File: ladder-core/crates/ladder-model/src/no_nc.rs
use serde::{Deserialize, Serialize};

/// Ported from org.grandviewtech.entity.enums.NoNc.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub enum NoNc {
    No,
    Nc,
    Default,
    Set,
    Reset,
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn wire_form_matches_original_java_enum_strings() {
        assert_eq!(serde_json::to_string(&NoNc::Set).unwrap(), "\"SET\"");
        assert_eq!(serde_json::to_string(&NoNc::Nc).unwrap(), "\"NC\"");
    }
}
